rollbar/lib/telemetry.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import threading
+import time
+
+import rollbar
+
+
+class Queue:
+    def __init__(self, size):
+        self.size = size
+        self.items = []
+        self.lock = threading.Lock()
+
+    def put(self, item):
+        with self.lock:
+            if len(self.items) >= self.size:
+                self.items = self.items[1:]
+            self.items.append(item)
+
+    def get_items(self):
+        with self.lock:
+            return self.items
+
+
+TELEMETRY_QUEUE_SIZE = 50
+TELEMETRY_QUEUE = Queue(TELEMETRY_QUEUE_SIZE)
+
+
+class TelemetryLogHandler(logging.Handler):
+    def __init__(self, formatter=None):
+        super(TelemetryLogHandler, self).__init__()
+        self.formatter = formatter
+
+    def emit(self, record):
+        self.setFormatter(self.formatter)
+        msg = {'message': self.format(record)}
+        data = {'body': msg, 'source': 'client', 'timestamp_ms': int(time.time()), 'type': 'log',
+                'level': record.levelname}
+
+        TELEMETRY_QUEUE.put(data)
+
+
+def set_log_telemetry(log_formatter=None):
+    logging.getLogger().addHandler(TelemetryLogHandler(log_formatter))
+
+
+def request(request_function, enable_req_headers, enable_response_headers):
+    def telemetry(*args, **kwargs):
+
+        def clean_headers(headers):
+            for h in headers:
+                if h in rollbar.SETTINGS['scrub_fields']:
+                    del headers[h]
```

Comment:
Maybe better here to set a scrubbed/redacted value (`headers[h] = '******'`) and preserve that the header was present.